src/lib/sort/sort.spec.ts

```diff
@@ -0,0 +1,263 @@
+import {async, ComponentFixture, TestBed} from '@angular/core/testing';
+import {Component, ElementRef, ViewChild} from '@angular/core';
+import {MdSort, MdSortHeader, Sort, SortDirection, MdSortModule} from './index';
+import {CdkDataTableModule, DataSource, CollectionViewer} from '../core/data-table/index';
+import {Observable} from 'rxjs/Observable';
+import {dispatchMouseEvent} from '../core/testing/dispatch-events';
+import {
+  getMdSortDuplicateMdSortableIdError,
+  getMdSortHeaderMissingIdError,
+  getMdSortHeaderNotContainedWithinMdSortError
+} from './sort-errors';
+import {wrappedErrorMessage} from '../core/testing/wrapped-error-message';
+import {map} from '../core/rxjs/index';
+
+describe('MdSort', () => {
+  let fixture: ComponentFixture<SimpleMdSortApp>;
+
+  let component: SimpleMdSortApp;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [MdSortModule, CdkDataTableModule],
+      declarations: [
+        SimpleMdSortApp,
+        CdkTableMdSortApp,
+        MdSortHeaderMissingMdSortApp,
+        MdSortDuplicateMdSortableIdsApp,
+        MdSortableMissingIdApp
+      ],
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(SimpleMdSortApp);
```

Comment:
`createComponent` has to go in a separate `beforeEach` block because `compileComponents()` is asynchronous